### ISSUE TITLE
Prediction model returns the inputed photo with clouds

### ISSUE THREAD

**Issue by RMNT** [ameraner/dsen2-cr#6]
After installing said libraries and formatting csv file as it's described in the readme file I managed to make the code run, however, the predicted output is exactly like the photo with cloud I have inputted. Has anyone else encountered something similar? What should I do to get the actual result? Thank you

**Comment by ameraner** [comment]
Hi @RMNT , thank you for your feedback. This is strange. One thing I can think of is that somehow you're training using the cloudy images as target. Are you sure that your csv setup/column order is correct?

**Comment by animecomico** [comment]
Test with this environment instructions on Anaconda, working for me

conda create -y -n dsen2_cr_v1 python=3.7.0
conda activate dsen2_cr_v1
conda install numpy=1.17.5
pip install tensorflow==1.13.2
pip install keras==2.2.4
pip install matplotlib==2.1.1
pip install rasterio
pip install pydot
pip install h5py
pip install graphviz
pip install scipy
pip install protobuf==3.6.1

**Comment by RMNT** [comment]
> Hi @RMNT , thank you for your feedback. This is strange. One thing I can think of is that somehow you're training using the cloudy images as target. Are you sure that your csv setup/column order is correct?

Thank you for your reply. I am using provided weights (full DSen2-CR model trained by optimizing the CARL loss) and am only testing the model. I double checked the csv file column order and it is [sample type, sentinel-1, sentinel-2 cloud free, sentinel-2 with clouds and the final one is file name]. Example: 3,s1/,noclouds/,clouds/,p14617.tif

**Comment by ameraner** [comment]
Sorry, then I'm not sure what is happening...

**Comment by StrangeReseach100PK** [comment]
Hi @ameraner  , I am also not able to see the output as expected. There is also some cloud on the prediction. Could you please let me know how many epochs you ran and what is your datasize? I just want to whether your model trained properly or not?  

Thank you

**Comment by ameraner** [comment]
Closing this as the questions have been answered, feel free to reopen if you have any more issues/questions!
